src/seedsigner/views/seed_views.py

```diff
@@ -2299,3 +2309,330 @@ def run(self):
+            self.controller.storage.discard_pending_shamir_share_set()
+            seed_num = self.controller.storage.finalize_pending_seed()
+            return Destination(SeedOptionsView, view_args={"seed_num": seed_num}, clear_history=True)
+
+        elif button_data[selected_menu_num] == self.PASSPHRASE:
+            return Destination(SeedAddSlip39PassphraseView)
+
+
+
+class SeedAddSlip39PassphraseView(View):
+    def __init__(self, initial_keyboard: str = seed_screens.SeedAddPassphraseScreen.KEYBOARD__LOWERCASE_BUTTON_TEXT):
+        super().__init__()
+        self.initial_keyboard = initial_keyboard
+        self.seed = self.controller.storage.get_pending_seed()
+
+
+    def run(self):
+        ret_dict = self.run_screen(
+            seed_screens.SeedAddPassphraseScreen, 
+            passphrase=self.seed.passphrase, 
+            title = _("SLIP-39 Passphrase"),
+            initial_keyboard=self.initial_keyboard,
+        )
+
+        # The new passphrase will be the return value; it might be empty.
+        self.seed.set_passphrase(ret_dict["passphrase"])
+
+        if "is_back_button" in ret_dict or len(self.seed.passphrase) == 0:
+                return Destination(SeedAddSlip39PassphraseExitDialogView)
+        
+        else:
+            self.controller.storage.convert_pending_shamir_share_set_to_pending_seed(passphrase=ret_dict["passphrase"], finalize=False)
+            return Destination(SeedReviewSlip39PassphraseView)
+
+
+
+class SeedAddSlip39PassphraseExitDialogView(View):
+    EDIT = ButtonOption("Edit passphrase")
+    DISCARD = ButtonOption("Discard passphrase", button_label_color="red")
+    SKIP = ButtonOption("Skip passphrase")  # NOT red since we're not throwing anything away
+
+    def __init__(self):
+        super().__init__()
+        self.seed = self.controller.storage.get_pending_seed()
+
+
+    def run(self):
+        if self.seed.passphrase:
+            title = _("Discard passphrase?")
+            message = _("Your current passphrase entry will be erased.")
+            button_data = [self.EDIT, self.DISCARD]
+        else:
+            title = _("Skip passphrase?")
+            message = _("You have not entered a passphrase yet.")
+            button_data = [self.EDIT, self.SKIP]
+        
+        selected_menu_num = self.run_screen(
+            WarningScreen,
+            title=title,
+            status_headline=None,
+            text=message,
+            show_back_button=False,
+            button_data=button_data,
+        )
+
+        if button_data[selected_menu_num] == self.EDIT:
+            return Destination(SeedAddSlip39PassphraseView)
+
+        elif button_data[selected_menu_num] in [self.DISCARD, self.SKIP]:
+            self.controller.storage.convert_pending_shamir_share_set_to_pending_seed(passphrase="", finalize=False)
+            return Destination(SeedShamirShareFinalizeView)
+
+
+
+class SeedReviewSlip39PassphraseView(View):
```

Comment:
Same here. The only changes from the bip39 equivalent are some of the cleanup and routing at the end.